src/main/java/com/chimbori/crux/articles/PostprocessHelpers.java

```diff
@@ -71,30 +75,62 @@
   private static final Set<String> RETAIN_TAGS_TOP_LEVEL = new HashSet<>(Arrays.asList(
       "p", "h1", "h2", "h3", "h4", "h5", "h6", "blockquote", "li"
   ));
-
+  
   static Document postprocess(Element topNode) {
     Log.i("postprocess");
     Document doc = new Document("");
     if (topNode == null) {
       return doc;
     }
+    
+    Set<Node> keepers=Collections.newSetFromMap(
+        new IdentityHashMap<Node,Boolean>());
+    for(Element element : topNode.select("[crux-keep]"))
+        keepers.addAll(lineage(topNode, element));
+    
+    PostprocessHelpers helper=new PostprocessHelpers(keepers);
 
-    removeNodesWithNegativeScores(topNode);
-    replaceLineBreaksWithSpaces(topNode);
-    removeUnlikelyChildNodes(topNode);
-    removeTagsButRetainContent(topNode);
-    removeTagsNotLikelyToBeParagraphs(topNode);
-    removeTopLevelTagsNotLikelyToBeParagraphs(topNode);
-    removeShortParagraphs(topNode);
-    removeDisallowedAttributes(topNode);
+    helper.removeNodesWithNegativeScores(topNode);
```

Comment:
Good question! I made them instance methods (as opposed to static methods) because:

* The methods were `private`, so changing them from static methods to instance methods would not affect any external callers.
* I needed to make the `keepers` set available to them, but I didn't want to change their signatures, so I added `keepers` as an instance field and made them instance methods.

Reply:
I see, thanks! Makes sense.